hub_sdk/base/server_clients.py

```diff
@@ -163,6 +163,42 @@ def _handle_signal(self, signum, frame) -> None:
         sys.exit(signum)
 
 
+class ProjectUpload(APIClient):
+    def __init__(self, headers):
+        """
+        Initialize the class with the specified headers.
+        Args:
+            headers: The headers to use for API requests.
+        """
+        super().__init__(f"{HUB_API_ROOT}/v1/projects", headers)
+        self.name = "project"
+
+    def upload_image(self, id: str, file):
+        """
+        Upload a project file to the hub.
+        Args:
+            id (YourIdType): The ID of the dataset to upload.
+            file (str): The path to the dataset file to upload.
+        Returns:
+            Any: The response from the upload request.
+        """
+        base_path = os.getcwd()
```

Comment:
@hassaanfarooq01 What happens if the user passes in the full path? Its a better idea to use pathlib.